src/gui/editors/SongEditor.cpp

```diff
@@ -293,6 +300,75 @@ void SongEditor::scrolled( int new_pos )
 
 
 
+void SongEditor::updateRubberband()
+{
+	if (rubberBandActive())
+	{
+		//take care of the zooming
+		int originX = m_origin.x();
+		if (m_currentZoomingValue != m_zoomingModel->value())
+		{
+			originX = m_trackHeadWidth + (originX - m_trackHeadWidth)
+					* m_zoomLevels[m_zoomingModel->value()] / m_zoomLevels[m_currentZoomingValue];
+		}
+
+		//take care of the scrollbar position
+		int hs = (m_leftRightScroll->value() - m_scrollPos.x()) * pixelsPerTact();
+		int vs = contentWidget()->verticalScrollBar()->value() - m_scrollPos.y();
+
+		//the adjusted origin point
+		QPoint origin = QPoint(qMax(originX - hs, m_trackHeadWidth), m_origin.y() - vs);
+
+		//paint the rubber band rect
+		rubberBand()->setGeometry(QRect( origin,
+										 contentWidget()->mapFromParent(QPoint(m_mousePos.x(), m_mousePos.y()))
+										).normalized());
+
+		//the index of the TrackView the mouse is hover
+		const TrackView * tv = trackViewAt(m_mousePos.y() - m_timeLine->height());
+		int rubberBandTrackview = 0;
+		if (tv)
+		{
+			for (auto it : trackViews())
+			{
```

Comment:
Can't understand the reason behind this loop; why won't you use `indexOf`?

Reply:
I understand. We could add `indexOfTrackView(const TrackView)` with `std::distance(std::find(m_trackViews.begin(), m_trackViews.end(), tv), m_trackViews.begin()` instead.